tuna/miopen/scripts/report.py

```diff
@@ -0,0 +1,251 @@
+  """Print tuning summary"""
+
+  #remove entries where sess or gv dont have an entry for the same config
+  dfr = dfr.loc[~dfr[['solver_x', 'solver_y']].isna().any(axis=1)].copy()
+  #compute ktime diff for fastest solver/config
+  dfr['diff'] = dfr['ktime_x'] - dfr['ktime_y']
+
+  #prct configs faster or slower
+  prct_positive = (dfr['diff'] > 0).sum() / dfr.shape[0] * 100
+  prct_equal = (dfr['diff'] == 0).sum() / dfr.shape[0] * 100
+  prct_negative = (dfr['diff'] < 0).sum() / dfr.shape[0] * 100
+  #pylint: disable=logging-format-truncated
+  LOGGER.info("configs with faster kernel_time: %f %%", round(prct_negative, 4))
+  LOGGER.info("configs with equal kernel_time: %f %%", round(prct_equal, 4))
+  LOGGER.info("configs with slower kernel_time: %f %% \n",
+              round(prct_positive, 4))
+
+  #averages
+  avg_positive = (dfr['diff'] > 0).mean()
+  avg_negative = (dfr['diff'] < 0).mean()
+
+  LOGGER.info("Mean for configs with faster kernel_time: %s %%", avg_negative)
+  LOGGER.info("Mean for configs with slower kernel_time: %s %%", avg_positive)
+  LOGGER.info("Mean for all configs: %s %%", dfr['diff'].mean())
+
+  dfr['%speedup'] = (dfr['ktime_x'] - dfr['ktime_y']) / (
+      (dfr['ktime_x'] + dfr['ktime_y']) / 2) * 100
+  LOGGER.info("Overall speed-up: %s %%", round(dfr['%speedup'].mean(), 4))
+
+  config_data = f"config_data_sess{args.session_id}_gv{args.golden_v}.csv"
+  LOGGER.info("Config report has been written to file: %s\n", config_data)
+  dfr.to_csv(config_data)
+
+
+def solver_report(args, dfr):
+  """Print detailed tuning analysis"""
+
+  LOGGER.info('Detailed solver report:')
+
+  #dataframe with fastest solvers(solver_x, solver_y)
+  df_compare = dfr.replace(-1, np.nan).groupby('config')[['ktime_x',
+                                                          'ktime_y']].idxmin()
+  df_compare.columns = ['idx_x', 'idx_y']
+
+  df_compare['solver_x'] = df_compare['idx_x'].apply(dfr['solver'].get)
+  df_compare['solver_y'] = df_compare['idx_y'].apply(dfr['solver'].get)
+
+  df_compare['ktime_x'] = df_compare['idx_x'].apply(dfr['ktime_x'].get)
+  df_compare['ktime_y'] = df_compare['idx_y'].apply(dfr['ktime_y'].get)
+
+  #dataframe where solvers have changed
+  dfr_diff_solvers = df_compare.loc[df_compare['solver_x'].ne(
+      df_compare['solver_y'])]
+  #dataframe where solvers are the same
+  dfr_same_solvers = df_compare.loc[df_compare['solver_x'].eq(
+      df_compare['solver_y'])]
+  dfr_same_solvers = dfr_same_solvers.drop(columns=['idx_x', 'idx_y'])
+  dfr_same_solvers['%diff'] = (dfr_same_solvers['ktime_x'] - dfr_same_solvers['ktime_y'])\
+                    / ((dfr_same_solvers['ktime_x'] + dfr_same_solvers['ktime_x'])/2) * 100
+  _, id_solver_map = get_id_solvers()
+  dfr_same_solvers['solver_x'] = dfr_same_solvers['solver_x'].apply(
+      id_solver_map.get)
+  dfr_same_solvers['solver_y'] = dfr_same_solvers['solver_y'].apply(
+      id_solver_map.get)
+  LOGGER.info('Mean %%change for same solver: %s',
```

Comment:
You are only looking at fastest solvers. Maybe change this comment to reflect that. But what I'm not sure about is why look at only the fastest "same" solvers? If you want to see how much a solver regressed going from golden_data to session_data, I'd do:

1. Only look at entries where I have access to kernel times for both solver_x and solver_y:
```
df = dfr.replace(-1, np.nan)
df = df[~df.ktime_x.isna()]
df = df[~df.ktime_y.isna()]
```
2. Find the average ktime_x and ktime_y for each solver
```
df = df.groupby("solver").mean()
``` 
3. Compute the avg. speedup and report:
```
avg_speedup = (df.ktime_y.mean() - df.ktime_x.mean()) / df.ktime_y.mean()

if avg_speedup < 0:
    LOGGER.info('On avg., the solvers in session_data are {abs(avg_speedup)}%% slower')
elif avg_speedup > 0:
    # report % speed up
else:
    # report no change
```

Reply:
I spoke to @cderb, for now we are mostly interested in fastest solvers speed-up, so i will update the comment to reflect that.